[DotNet][DateTime] Refactor GetModAndDate method since it has too many parameters

ceciliaavila opened the pull request:
- Created new class ModAndDateResult containing the properties we need to return in the method GetModAndDate
- Redesigned GetModAndDate method to remove the out/ref parameters and return an instance of a ModAndDateResult object.

VictorGrycuk commented on .NET/Microsoft.Recognizers.Text.DateTime/Utilities/GetModAndDateResult.cs:
```diff
@@ -0,0 +1,33 @@
+
+namespace Microsoft.Recognizers.Text.DateTime.Utilities
+{
+    public class GetModAndDateResult
```
I would like to suggest `ModAndDateResult` as name for the class. Usually only non void methods start with `Get`.

ceciliaavila replied:
Feedback applied!

JuanAr replied:
I think @VictorGrycuk referred to rename only the class. You also renamed the method, which I'd roll back.

JuanAr replied:
`public class ModAndDateResult` and `public ModAndDateResult GetModAndDate(`

ceciliaavila replied:
Done!

ceciliaavila committed:
Refactor BaseDatePeriodParser - GetModAndDate method since it has too many out/ref parameters

- Created new class GetModAndDateResult containing the properties we need to return in the method GetModAndDate
- Redesigned GetModAndDate method to remove the out/ref parameters and return an instance of a GetModAndDateResult object.

ceciliaavila committed:
Renamed class 'GetModAndDateResult' to 'ModAndDateResult'

JuanAr reviewed (approved):
👍
